### PR TITLE
fix(event-explorer): prevent autoload duplicates

### PR DESCRIPTION
**Pull request by thmsobrmlr**
## Problem

The event explorer duplicates events when the autoload toggle is switched on.

Solves duplicates issue of https://github.com/PostHog/posthog/issues/15143.

## Changes

The reason for this seems to be that `visit_datetime` in `escape_sql.py` uses a format without milliseconds i.e. `2023-04-17T22:15:38.843Z` would become `2023-04-17T22:15:38Z`. It is necessary to use `toDateTime64` instead of `toDateTime` in order for ClickHouse to accept the milliseconds as well.

## How did you test this code?

Not yet

### CODE REVIEW

**Comment by thmsobrmlr** [comment]
@mariusandra This is probably not the right solution for this problem. Mind giving a look and letting me know what you'd suggest?

File: posthog/hogql/escape_sql.py [hunk 1]
```diff
@@ -104,10 +104,10 @@ def visit_fakedatetime(self, value: datetime):
         return self.visit_datetime(value)
 
     def visit_datetime(self, value: datetime):
-        datetime_string = value.astimezone(ZoneInfo(self._timezone)).strftime("%Y-%m-%d %H:%M:%S")
+        datetime_string = value.astimezone(ZoneInfo(self._timezone)).strftime("%Y-%m-%d %H:%M:%S.%f")
         if self._dialect == "hogql":
             return f"toDateTime({self.visit(datetime_string)})"  # no timezone for hogql
-        return f"toDateTime({self.visit(datetime_string)}, {self.visit(self._timezone)})"
+        return f"toDateTime64({self.visit(datetime_string)}, 3, {self.visit(self._timezone)})"
```
Comment (thmsobrmlr) [inline posthog/hogql/escape_sql.py:110]:
```suggestion
        return f"toDateTime64({self.visit(datetime_string)}, 6, {self.visit(self._timezone)})"
```

**Review by mariusandra — APPROVED** [review]
I think this actually makes sense. Supporting timestamps with milliseconds is something I'd take for granted, so let's just convert them all to `dateTime64`.

**Comment by thmsobrmlr** [comment]
Thanks for the additions 🙌

**Pull request merged**